### PR TITLE
Add unit test for kafka partition ack manager

### PR DESCRIPTION
**Pull request by neil-xie**
**What changed?**
Add unit test for kafka partition ack manager
Refactor complete message method

**Why?**

**How did you test it?**
Unit test

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: common/messaging/kafka/consumerImpl.go [hunk 1]
```diff
@@ -210,7 +210,13 @@ func (h *consumerHandlerImpl) completeMessage(message *messageImpl, isAck bool)
 				tag.KafkaOffset(message.Offset()))
 		}
 	}
-	ackLevel := h.manager.CompleteMessage(message.Partition(), message.Offset(), isAck)
+	ackLevel, err := h.manager.CompleteMessage(message.Partition(), message.Offset(), isAck)
+	if err != nil {
+		h.logger.Error("complete an message that hasn't been added",
```
Comment (taylanisikdemir) [inline common/messaging/kafka/consumerImpl.go:215]: maybe a more descriptive message "Failed to complete message"

File: common/messaging/kafka/consumerImpl.go [hunk 2]
```diff
@@ -210,7 +210,13 @@ func (h *consumerHandlerImpl) completeMessage(message *messageImpl, isAck bool)
 				tag.KafkaOffset(message.Offset()))
 		}
 	}
-	ackLevel := h.manager.CompleteMessage(message.Partition(), message.Offset(), isAck)
+	ackLevel, err := h.manager.CompleteMessage(message.Partition(), message.Offset(), isAck)
+	if err != nil {
```
Comment (taylanisikdemir) [inline common/messaging/kafka/consumerImpl.go:214]: thanks for fixing this

File: common/messaging/kafka/partitionAckManager_test.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
+// The MIT License (MIT)
```
Comment (taylanisikdemir) [inline common/messaging/kafka/partitionAckManager_test.go:1]: let's also rename the touched files to snake_case (or the whole folder).

File: common/messaging/kafka/partitionAckManager_test.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	// Setup
+	metricsClient := metrics.NewClient(tally.NoopScope, metrics.History)
+	logger := testlogger.New(t)
+	pam := newPartitionAckManager(metricsClient, logger)
+
+	partitionID := int32(1)
+	messageID := int64(100)
+	
+	// Add a message first to simulate a real-world scenario, this will create ackMgr for partition
+	pam.AddMessage(partitionID, messageID)
```
Comment (taylanisikdemir) [inline common/messaging/kafka/partitionAckManager_test.go:62]: let's move these to test body at line 100 so each test case is isolated and don't have a chance to impact each other

**Review by taylanisikdemir — APPROVED** [review]
LGTM with comments

File: common/messaging/kafka/partitionAckManager_test.go (hunk 3 above)
Comment (neil-xie) [reply common/messaging/kafka/partitionAckManager_test.go:1]: Sure, I will rename the other files when I write test for them.

**Pull request merged**